cmd/lakefs/cmd/run.go

```diff
@@ -112,27 +113,60 @@ var runCmd = &cobra.Command{
 
 		// validate service names and turn on the right flags
 		dbParams := cfg.GetDatabaseParams()
-		dbPool := db.BuildDatabaseConnection(ctx, dbParams)
-		defer dbPool.Close()
-		if err := db.ValidateSchemaUpToDate(ctx, dbPool, dbParams); errors.Is(err, db.ErrSchemaNotCompatible) {
-			logger.WithError(err).Fatal("Migration version mismatch, for more information see https://docs.lakefs.io/deploying-aws/upgrade.html")
-		} else if errors.Is(err, migrate.ErrNilVersion) {
-			logger.Debug("No migration, setup required")
-		} else if err != nil {
-			logger.WithError(err).Warn("Failed on schema validation")
-		}
 
-		lockdbPool := db.BuildDatabaseConnection(ctx, dbParams)
-		defer lockdbPool.Close()
+		var (
+			kvStore    kv.Store
+			kvParams   params.KV
+			dbPool     db.Database
+			lockDBPool db.Database
+			err        error
+		)
+		if dbParams.KVEnabled {
+			kvParams = cfg.GetKVParams()
+			kvStore, err = kv.Open(ctx, kvParams)
+			if err != nil {
+				logger.WithError(err).Fatal("Failed to open KV store")
+			}
+			defer kvStore.Close()
 
+			// Check if migration required only on postgres
+			migrationRequired := false
+			if dbParams.Type == kvpg.DriverName {
```

Comment:
Does this mean that every DynamoDB installation will work even if no migration was made?

Reply:
After reading the code of `ValidateSchemaVersion` there are few things I don't understand. Where can I go to to understand more about the migration process?

Reply:
This code handles the specific checks for migration from DB to KV. It will change on post KV migration.
It basically checks that if we are using postgres - that we do it only after we migrated the data or alternatively start with a fresh install.
Since we do not support migration directly to dynamodb, it is up to the user to make sure he doesn't try to upgrade a previous lakeFS version without going through the proper migration process via Postgres. This will be documented in the upgrade guide